iwyu_tool.py

```diff
@@ -25,9 +25,67 @@
 import json
 import argparse
 import subprocess
-
-
-def run_iwyu(cwd, compile_command, iwyu_args, verbose):
+import re
+
+
+CORRECT_RE = re.compile(r'^\((.*?) has correct #includes/fwd-decls\)$')
+SHOULD_ADD_RE = re.compile(r'^(.*?) should add these lines:$')
+SHOULD_REMOVE_RE = re.compile(r'^(.*?) should remove these lines:$')
+FULL_LIST_RE = re.compile(r'The full include-list for (.*?):$')
+END_RE = re.compile(r'^---$')
+LINES_RE = re.compile(r'.*// lines ([0-9]+)-[0-9]+$')
+
+
+GENERAL, ADD, REMOVE, LIST = range(4)
```

Comment:
I think this needs to be `list(range(4))` to be Python 3-compatible (quirky but nice idiom for enums, btw, I've never seen that before)